content/en/docs/tutorials/configuration/updating-configuration-via-a-configmap.md

```diff
@@ -0,0 +1,723 @@
+```
+NAME                                          READY   STATUS    RESTARTS   AGE
+immutable-configmap-volume-5c677f6494-n7cg5   1/1     Running   0          30s
+immutable-configmap-volume-5c677f6494-sdhkp   1/1     Running   0          30s
+immutable-configmap-volume-5c677f6494-vsszf   1/1     Running   0          30s
+```
+
+The Pod's container refers to the data defined in the ConfigMap and uses it to print a report to stdout. You can check this report by viewing the logs for one of the Pods in that Deployment:
+```shell
+# Pick one Pod that belongs to the Deployment, and view its logs
+kubectl logs deployments/immutable-configmap-volume
+```
+
+You should see an output similar to:
+```
+Found 3 pods, using pod/immutable-configmap-volume-5c677f6494-n7cg5
+Tue Mar  5 11:38:27 UTC 2024 My preferred season is summer
+Tue Mar  5 11:38:37 UTC 2024 My preferred season is summer
+Tue Mar  5 11:38:47 UTC 2024 My preferred season is summer
+Tue Mar  5 11:38:57 UTC 2024 My preferred season is summer
+Tue Mar  5 11:39:07 UTC 2024 My preferred season is summer
+```
+
+Edit the ConfigMap:
+```shell
+kubectl edit configmap season
+```
+
+In the editor that appears, change the value of key `season` from `summer` to `winter`.
+
+Here's an example of how that manifest could look after you edit it:
+
+```yaml
+apiVersion: v1
+data:
+  season: winter
+immutable: true
+kind: ConfigMap
+# You can leave the existing metadata as they are.
+# The values you'll see won't exactly match these.
+metadata:
+  annotations:
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"v1","data":{"season":"summer"},"immutable":true,"kind":"ConfigMap","metadata":{"annotations":{},"name":"season","namespace":"default"}}
+  creationTimestamp: "2024-03-05T11:36:22Z"
+  name: season
+  namespace: default
+  resourceVersion: "3840"
+  uid: 10cdc5e6-47b8-4473-a9f3-ed401c2073f9
+```
+
+Attempt to save the changes. You should see that the save fails and a temporary manifest is opened which contains the reason for failure:
+```
+# reopened with the relevant failures.
+#
+# configmaps "season" was not valid:
+# * data: Forbidden: field is immutable when `immutable` is set
+#
+```
+
+Without any further change, save the temporary manifest. You should see an output similar to:
+```
+A copy of your changes has been stored to "/var/folders/nn/bb6y1j8d69n9lhqppjlrfthm0000gn/T/kubectl-edit-139284375.yaml"
+error: Edit cancelled, no valid changes were saved.
+```
+
+Once a ConfigMap is marked as immutable, it is not possible to revert this change nor to mutate the contents of the data or the binaryData field.  
+In order to modify the contents, the only option is to delete and recreate the ConfigMap.
+
+Delete and recreate the ConfigMap `season` by using the temporary manifest as follows:
+```shell
+# Use the appropriate path to the temporary manifest
+kubectl replace --force -f /var/folders/nn/bb6y1j8d69n9lhqppjlrfthm0000gn/T/kubectl-edit-139284375.yaml
+```
```

Comment:
This isn't always enough, because the kubelet doesn't watch for changes to immutable ConfigMaps.

Instead, I would put a date in the ConfigMap name, and teach switching to referencing a new different ConfigMap.

For example:
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: "company-name-20150801"
data:
  company_name: "ACME, Inc." # existing fictional company name
immutable: true
```

and then switch to something like:
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: "company-name-20240312"
data:
  company_name: "Fiktivesunternehmen GmbH" # new fictional company name
immutable: true
```

Reply:
This is implemented in the latest commit.